### DISCUSSION TITLE
How to create a dynamic table?

### DISCUSSION THREAD

**Discussion by prateekgoel** [vitusortner/floor#579] [Q&A]
In my flutter application I have a usecase wherein I need to store the user data in separate tables for each user. How can I create multiples tables like user_[userid] ?

I have been through the documentation and Github issues, but couldn't find anything on this.

@vitusortner Any suggestions on this please?

**Comment by mqus** [comment]
This is not possible with floor, floor assumes a fixed schema(with migrations). But depending on the amount of users and your usecase you could create a separate database per user and name the databases (via the builder) according to their IDs. Common data (e.g. the user ID to User mapping) is then put into a separate "common" database. But this does mean that you may not be able to join the data across those databases.

Another (more relational) way would be to put the userid as a separate column in all relevant tables and put `userid == :userid` in the WHERE clause to distinguish between the data.

**Reply by prateekgoel** [reply]
Thanks

**Reply by prateekgoel** [reply]
For now I am using rawquery to achieve my requirement.
